docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1021 @@
+# encoding: utf-8
+----------------------------------
+
+to the top of your {plugintype} plugin. Logstash depends on things being in
+UTF-8, so we put this here to tell the Ruby interpreter that we’re going to be
+using the UTF-8 encoding.
+
+==== `require` Statements
+
+A Logstash {plugintype} plugin requires some parent classes that can be
+referenced through the indicated statements. The following require statements
+are mandatory:
+
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+require "logstash/{plugintype}s/base"
+require "logstash/namespace"
+----------------------------------
+
+Of course, the plugin you build may depend on other code, or even gems. Just put
+them here along with these Logstash dependencies.
+
+=== Plugin Body
+
+Let's go through the various elements of the plugin itself.
+
+==== Inline Documentation
+Logstash provides infrastructure to automatically generate documentation for
+plugins. We use the asciidoc format to write documentation so _any_ comments in
+the source code will be first converted into asciidoc and then into html.
+
+All plugin documentation is then rendered and placed in
+http://www.elasticsearch.org/guide/latest/logstash[the Logstash section of the Elasticsearch Guide].
+
+The inline documentation can include code blocks and config examples! To include
+ruby code, use the asciidoc `[source,ruby]` directive:
+
+[source,txt]
+[subs="attributes"]
+----------------------------------
+# Using hashes:
+# [source,ruby]
+# ----------------------------------
+# match => {
+#  "field1" => "value1"
+#  "field2" => "value2"
+#  ...
+# }
+# ----------------------------------
+----------------------------------
+
+In the rendered html document, this block would look like:
+
+[]
+=========================
+Using hashes:
+[source,ruby]
+----------------------------------
+match => {
+  "field1" => "value1"
+  "field2" => "value2"
+  ...
+ }
+----------------------------------
+=========================
+
+
+TIP: For more asciidoc formatting tips, see the excellent reference here:
+https://github.com/elasticsearch/docs#asciidoc-guide
+
+==== `class` Declaration
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+class LogStash::{pluginclass}::{pluginnamecap} < LogStash::{pluginclass}::Base
+----------------------------------
+
+The class name will need to be a sub-class of
+`LogStash::pass:attributes[{pluginclass}]::Base`, and it should extend it as
+shown in the line above. You should name your class in a way that closely
+mirrors the plugin name, e.g.
+`LogStash::pass:attributes[{pluginclass}]::pass:attributes[{pluginnamecap}]` in
+this case.
+
+==== The `config_name`
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+  config_name "{pluginname}"
+----------------------------------
+This is the name your plugin will call inside the appropriate block.
+
+For this example, if we set `config_name "pass:attributes[{pluginname}]"` in our
```

Comment:
In this case, it really should be "you" instead of "we", "if you set...". In general, I'm not a fan of using "we", but the ES style is very much "Let's take a look at this together". That's fine, but where it's really the reader who has to do something, I still think we should address them directly instead of inserting us into the action as a silent partner.